### ISSUE TITLE
The shipped rating-ladder selection costs about two days per checkpoint

### ISSUE THREAD

**Issue by mmmfrieddough** [mmmfrieddough/anthro-chess#167]
## What was measured

While taking the first real readings through `anthro eval suite` (#162), the rating ladder turned out to be far more expensive than the rest of the harness combined.

`configs/evaluation/rating-ladder.toml` declares 4 ratings × 3 temperatures plus 3 ablated seats = **15 seats**, which is 105 pairings. At 16 openings, both color assignments, 1 game per position and 3 seeds, that is:

```
105 pairings × 16 openings × 2 colors × 1 game/position × 3 seeds = 10,080 games
```

Measured generation cost on Apple Silicon MPS against `training-blitz-30k-v4` step 8000 is roughly **15–18 s/game** — from `anthro eval rollout` at 36 games in 558 s, and a 4-game ladder in 71 s. That puts the shipped selection at **42–50 hours for one checkpoint**.

The extrapolation is from small samples and should be treated as an order of magnitude, not a figure. The game count is exact.

## Why it matters

- It is very likely this selection has never been executed at its declared size. The ladder landed in #63 and #146 has not run yet.
- #146 needs two checkpoints, which is on the order of four days of ladder alone.
- In #162 the ladder had to be excluded from the reduced sweep entirely, because its cost is a seat grid rather than a sample size and its sample dials (seeds, openings) are already near their floor while it is still playing hundreds of games. That leaves it in the full sweep, where the number above is what "full" means.

## Why the obvious lever is not available

Cost is quadratic in seat count and the seats *are* the measurement. Worse, `docs/decisions/0022-one-joint-rating-ladder-fit.md` makes one joint fit load-bearing: every seat is placed on a single internal scale, so a ladder fitted on a different set of seats cannot be read against this one at all. Dropping seats does not produce a less precise version of this measurement; it produces a different, incomparable one.

So this is a deliberate decision about what the ladder's declared size should be, not a tuning exercise.

## Directions worth weighing

- **Accept and schedule it.** Declare the ladder a rare, deliberately-scheduled reading rather than part of any routine sweep, and say so in `docs/evaluation.md`.
- **Cut the declared grid once, permanently.** Fewer temperatures or ratings gives a cheaper ladder for the rest of the project's life, at the cost of ending the current (empty) series. Nothing is protected before the core is designated, so the cut is free *now* and expensive later — which argues for deciding early rather than late.
- **Reduce games per pairing.** Openings and seeds are genuine sample sizes. 1 seed and 4 openings is 840 games (~4 h); 1 seed and 1 opening is 210 (~1 h). Both are legitimate reductions, but at 210 games a 105-pairing round robin averages 2 games per pairing, which may be too thin to fit.
- **Make generation cheaper.** 15–18 s/game for games that mostly hit the 300-ply limit suggests the per-decision cost in the generation path, not the model, may dominate. Worth measuring before accepting the number as fixed.

## Acceptance criteria

- The ladder's declared size is a recorded decision with its cost stated.
- If the grid changes, `configs/evaluation/rating-ladder.toml` and the affected docs change with it.
- If it stays, `docs/evaluation.md` says plainly that a full ladder is a scheduled reading rather than a routine one.

## References

- #162 — where the cost was measured
- #146 — the reading that will pay this cost first
- `docs/decisions/0022-one-joint-rating-ladder-fit.md`
- `docs/evaluation.md` — "The Implemented Ladder"

**Comment by mmmfrieddough** [comment]
Added #166 as a blocker, and this issue now blocks #146 (which #164 previously
blocked; #164 is closed as superseded by this issue, #168, and #166).

The reason #166 comes first is the question this issue already raises: whether
15–18 s/game is the real cost of generation or an artifact of the path.

Sampling a live `anthro eval run` on Apple Silicon put 693 of 1,300 main-thread
samples in `slot_nb_bool -> is_nonzero -> item() -> _local_scalar_dense_mps ->
MPSStream::synchronize -> [_MTLCommandBuffer waitUntilCompleted]`. That is Python
evaluating a tensor in boolean context, draining the command buffer before
continuing. One of the two call sites is the finite check on action logits in
`src/anthro_chess/inference/runner.py`, which runs on **every generated move** —
so it is on the ladder's path, not just the scoring path.

If that stall is a material part of the per-game cost, then the 42–50 hour figure
is partly a defect rather than a property of the seat grid. This issue proposes a
permanent cut, and argues correctly that the cut is free now and expensive later.
That argument cuts both ways: a permanent decision made from an inflated number
is also expensive later, and per
`docs/decisions/0022-one-joint-rating-ladder-fit.md` a ladder fitted on different
seats cannot be read against this one at all, so it is not a decision that can be
revisited cheaply.

Re-measuring after #166 should be quick — the 4-game ladder timing already cited
here is enough to tell whether the per-game cost moved.

Nothing about the game count changes: 105 pairings and 10,080 games are exact and
independent of how fast a game runs. If the cost per game barely moves, this
issue proceeds exactly as written with one uncertainty removed.

**Comment by mmmfrieddough** [comment]
The ladder has now been run at its declared size, twice, during #146. The
estimates this issue was filed on can be replaced with measurements.

**Measured cost: 2.01 h and 1.83 h per checkpoint** — 3.84 hours for both — for
the full 105 pairings and 10,080 games, on Apple Silicon MPS.

That is against the 42–50 hours estimated here. The difference is #166: the
per-move device synchronization this issue suspected ("15–18 s/game ... suggests
the per-decision cost in the generation path, not the model, may dominate. Worth
measuring before accepting the number as fixed") was real and is fixed. Measured
generation cost is now about 0.7 s/game.

The game count was exact and unchanged, as this issue said it would be.

**This removes the argument for cutting the grid.** A two-hour benchmark is a
scheduled reading, not a project-defining constraint, and per
`docs/decisions/0022-one-joint-rating-ladder-fit.md` a cut would end the current
series irreversibly. The "accept and schedule" direction listed here is now the
cheap one.

One caveat in the other direction, from #177: at these checkpoints the ladder's
headline quantities read as noise — spans of 10 to 17 Elo across a 900-Elo
configured range at both checkpoints, with ordering at chance. The quantity that
did discriminate is not in the headline: scored games rose from 3,405 to 5,273 of
10,080. Between 47% and 66% of games hit the ply limit, which is where most of
the two hours goes.

So the open question is no longer what the ladder costs. It is whether the
declared grid is measuring the right thing, which is a better question to spend a
permanent decision on.

**Comment by mmmfrieddough** [comment]
Registered as blocking #90 and #177.

**#90 — cut pool generation two and designate the evaluation core.** This is the deadline. Decision 0013 protects nothing before the core is designated, so a grid cut is free until #90 and permanently expensive after it.

There is a second reason the two belong together. `configs/evaluation/rating-ladder.toml` draws its openings from a pool view (`artifacts/lichess-blitz-2017-04-test`, view `ladder-openings`), and the config's own header lists the opening source as part of series identity. So #90 ends every ladder series regardless of what happens here. Deciding the grid first means one break at the seam instead of two, and it means the first ladder series that is actually protected starts at the size we intend to keep.

**#177 — the rating dial changes style but not strength.** Re-reading that finding means running the ladder. It should be run at whatever grid this issue settles on, not at the current one.

The practical consequence is that this should be decided early, before #89 and #90 consume the data work.

**Comment by mmmfrieddough** [comment]
Claimed for implementation by Claude Code (Opus 5).

Session: a0c2e8f0-863e-4c43-8858-f51625c28f40
